### ISSUE TITLE
My experience so far, questions about Improvements section

### ISSUE THREAD

**Issue by ManuGithubSteam** [daliansky/XiaoMi-Pro-Hackintosh#462]
Hi

First i want to thank you all for your hard work! 

I just got around to set up my hackintosh :-) It was a smooth experience so far. Touchpad worked at install and everything... Very impressed! And it is breathtaking to open the lid and have a mac to my disposal (i could never afford a real macbook myself) The most trouble was making a booting usb stick but i documented my adventures here:  https://github.com/ManuGithubSteam/XiaoMi-Pro-2018-HackintoshOC

I was able to get internal Wifi to work and now some more question arose. If you find this ticket later. I will document my notes on my processes on my wiki.

I will do the DVMT_and_0xE2_fix and would like to update the bios to XMAKB5R0P0906 to get the fan noise reduction. The scripts work for this version "as-is" correct ? I have the 150MX btw Just double checking because of bios...
Also out of curiosity is more than 64mb possible for the frambuffer ?

NVMeFix: I read that it did not so much good on some NVMEs. So im wondering: Is it save to use on a Samsung 970 Evo ?
Sadly there is no list of supported SSDs in the repo... I would inject it with OC like the itlwm kext correct ?

one-key-cpufriend (Link broken btw ) - What is the reason/thinking behind to use this ? I read the page but m not sure if i need or want it.  What SMBIOS is set in OC by default ? Does this correspond to the MAC serial number ? Mine is now: OPENCORE_SN1 so no real number at all. Not sure if this is ok...

Does igfxrpsc work on 10.15.7. ? 

Also for my idea with the sd card reader with vm solution: Is the SSDT-USB applied or ON to save power by default (the efi pack from the release page)?
Because if the SD Card reader is off, even if linux could detect it it would not as it is off.

How would i turn the patch off if this is the case to get the card reader "working" so linux may detect it ? Could i only turn off the fingerprint reader and let the sd-card reader have power ? 
Thanks for clarifying that :-)
 
Also how can boot into recovery mode ? Pressing CMD (Windows key)-R does noting on boot ?

Thank you all for your hard work and i hope i can contribute with the sd-card reader thing and some more notes on how to do things.

Thank you for reading.

**Comment by stevezhengshiqi** [comment]
> I will do the DVMT_and_0xE2_fix and would like to update the bios to XMAKB5R0P0906 to get the fan noise reduction. The scripts work for this version "as-is" correct ? I have the 150MX btw Just double checking because of bios...

The script works on all the version above and includes 0502, but no guarantee it's 100% safe.

> Also out of curiosity is more than 64mb possible for the frambuffer ?

Sure it can be larger, e.p. 256mb. But it would exceed the size what AppleIntelFrambuffer expects, which will affect the performance.

> NVMeFix: I read that it did not so much good on some NVMEs. So im wondering: Is it save to use on a Samsung 970 Evo ?
Sadly there is no list of supported SSDs in the repo... I would inject it with OC like the itlwm kext correct ?

No idea and I can't verify that. If you don't have any problems without NVMeFix, I recommend remaining unchanged.

> one-key-cpufriend (Link broken btw ) - What is the reason/thinking behind to use this ? I read the page but m not sure if i need or want it. 

Adjust low frequency mode (LFM) and EPP, you can google that. No recommend to do so if you have no idea what are they.

> What SMBIOS is set in OC by default ?

MacBookPro14,1. You can always check that by looking at the code in [config.plist](https://github.com/daliansky/XiaoMi-Pro-Hackintosh/blob/master/OC/config.plist#L1217). Same as serial numbers, you can add them in corresponding field in config.plist.

> Does igfxrpsc work on 10.15.7. ?

No idea since I am not a developer of [WhateverGreen](https://github.com/acidanthera/WhateverGreen). They just mentioned that `Disabled RPS control patch by default due to a bug in 10.15.6 IGPU drivers` in [Changelog.md](https://github.com/acidanthera/WhateverGreen/blob/f31896efc9eb28e0eaad694e9268c7ae5e6f9134/Changelog.md)

> Because if the SD Card reader is off, even if linux could detect it it would not as it is off.

All the changes made in macOS won't affect other systems. I restricted the functionality of all SSDTs exclusive to macOS by adding `If _OSI("Darwin)`.

> Also how can boot into recovery mode ? Pressing CMD (Windows key)-R does noting on boot ?

Not sure what bootloader you are using. Press `Space` if you are using OpenCore, and press Fn+F3 if you are using Clover.

Basically these questions are covered in README and code in this repository. You can have a closer look at config.plist and [OpenCorePkg document](https://github.com/acidanthera/OpenCorePkg/blob/master/Docs/Configuration.pdf) to find more functionality of the bootloader

**Comment by ManuGithubSteam** [comment]
Hi Steve,

Thanks as always.

With the SD-CardReader i mean:
When i boot MacOS is the SSDT-USB applied or ON to save power by **default** as this would mean that a virtual machine in wich i would run linux with VMware Fusion (to have hopefully the card reader functional) would not detect the reader because it has no power.

If that is so. How could i deactivate that patch to give it power ?

Thanks for taking the time :+1: Sadly the Readme is sometimes not so clear to me...

**Comment by stevezhengshiqi** [comment]
[SSDT-USB_SD.zip](https://github.com/daliansky/XiaoMi-Pro-Hackintosh/files/5354072/SSDT-USB_SD.zip)

**Comment by MarFre22** [comment]
Thanks for make your contribution to the community!

> NVMeFix: I read that it did not so much good on some NVMEs. So im wondering: Is it save to use on a Samsung 970 Evo ?
> Sadly there is no list of supported SSDs in the repo... I would inject it with OC like the itlwm kext correct ?
NVMeFix is used to improve the power consumption of drivers NVMe and, also, improve their compatibility with MacOS. Personally 970 evo it's a beast on windows (on linux I don't know), but on MacOS there are better options like Sabrent Rocket or Sabrent Rocket Q (in theses drives I don't know if NVMeFix is needed, but I have been using it without problems). On 970 evo, NVMeFIX is absolutely essential because it gets very hot and it's more power hungry that Sabrent.
Some info: [here](https://github.com/daliansky/XiaoMi-Pro-Hackintosh/issues/265#issuecomment-691532644), [here](https://github.com/daliansky/XiaoMi-Pro-Hackintosh/issues/381#issuecomment-616533216)

> one-key-cpufriend (Link broken btw ) - What is the reason/thinking behind to use this ? I read the page but m not sure if i need or want it.  What SMBIOS is set in OC by default ? Does this correspond to the MAC serial number ? Mine is now: OPENCORE_SN1 so no real number at all. Not sure if this is ok...
I'm always using cpufriend and it gives me a good battery time. I was able to use it by explore folders, find the shortcut folder on it and run the script. I don't know if it possible now.

**Comment by ManuGithubSteam** [comment]
OK so i guess this file has the SD-Card reader functioning ? Will try this out later after work :-)
I'm not a coder so i don't really understand the C code.... Sorry

> [SSDT-USB_SD.zip](https://github.com/daliansky/XiaoMi-Pro-Hackintosh/files/5354072/SSDT-USB_SD.zip)

**Comment by ManuGithubSteam** [comment]
> Thanks for make your contribution to the community!
> 
> > NVMeFix: I read that it did not so much good on some NVMEs. So im wondering: Is it save to use on a Samsung 970 Evo ?
> > Sadly there is no list of supported SSDs in the repo... I would inject it with OC like the itlwm kext correct ?
> 
> NVMeFix is used to improve the power consumption of drivers NVMe and, also, improve their compatibility with MacOS. Personally 970 evo it's a beast on windows (on linux I don't know), but on MacOS there are better options like Sabrent Rocket or even Sabrent Rocket Q (in theses drives I don't know if NVMeFix is needed, but I have been using it without problems). On 970 evo, NVMeFIX is absolutely essential because it gets very hot and it's more power hungry than Sabrent.
> Some info: [here](https://github.com/daliansky/XiaoMi-Pro-Hackintosh/issues/265#issuecomment-691532644), [here](https://github.com/daliansky/XiaoMi-Pro-Hackintosh/issues/381#issuecomment-616533216)
First: It is also a beast on Linux :-)
Thanks for letting me know. I will do the NVME Fix then. I was wondering why the battery is draining so quick when the laptop goes unter 20 Percent is is especially noticeabe but this seems to be local. Do you think i should even discard the EVO and buy a new Sabrent Rocket Q?  How us your  evo holding up ?
> 
> > one-key-cpufriend (Link broken btw ) - What is the reason/thinking behind to use this ? I read the page but m not sure if i need or want it.  What SMBIOS is set in OC by default ? Does this correspond to the MAC serial number ? Mine is now: OPENCORE_SN1 so no real number at all. Not sure if this is ok...
> 
> I'm always using cpufriend and it gives me a good battery time. I was able to use it by explore folders, find the shortcut folder on it and run the script. I don't know if it possible now.
Ok is it hard to set up ? Its seems worth the effort but the guides are all skeleton like a bit more info what wich part does what and how it fits all together would be nice...

**Comment by MarFre22** [comment]
> First: It is also a beast on Linux :-)

Oh nice, thanks for confirm that ;)


> Thanks for letting me know. I will do the NVME Fix then. I was wondering why the battery is draining so quick when the laptop goes unter 20 Percent is is especially noticeabe but this seems to be local. Do you think i should even discard the EVO and buy a new Sabrent Rocket Q?  How us your  evo holding up ?

Assuming your battery it's rock solid yet, I think, battery draining it's more visible without cpufriend. 
I had a 970 evo some time ago, due to using more MacOS than Windows (dualboot) I needed  to sell my 970 to buy an Rocket Q - I know 970 evo is a beast (MLC compared QLC, TBW, general speeds) but on Mac Sabrent Rocket Q has better performance due to the Phison controller (Samsung proprietary controllers have constant TRIM crash on Mac -> more heat). After I bought Sabrent Rocket Q, I have much better performance, huge improvement. If you can buy Sabrent Rocket (not Q), it's a more safe choice due to TBW and speeds compared with 970 evo.

> > > one-key-cpufriend (Link broken btw ) - What is the reason/thinking behind to use this ? I read the page but m not sure if i need or want it.  What SMBIOS is set in OC by default ? Does this correspond to the MAC serial number ? Mine is now: OPENCORE_SN1 so no real number at all. Not sure if this is ok...

> Ok is it hard to set up ? Its seems worth the effort but the guides are all skeleton like a bit more info what wich part does what and how it fits all together would be nice...

Only follow "How to install" on stevezhengshiqi/one-key-cpufriend.

**Comment by ManuGithubSteam** [comment]
> Assuming your battery it's rock solid yet, I think, battery draining it's more visible without cpufriend.
ok thanks will try the cpu friend then.
> I had a 970 evo some time ago, due to using more MacOS than Windows (dualboot) I needed to sell my 970 to buy an Rocket Q - I know 970 evo is a beast (MLC compared QLC, TBW, general speeds) but on Mac Sabrent Rocket Q has better performance due to the Phison controller (Samsung proprietary controllers have constant TRIM crash on Mac -> more heat). After I bought Sabrent Rocket Q, I have much better performance, huge improvement. If you can buy Sabrent Rocket (not Q), it's a most safest choice due to TBW and speeds compared with 970 evo.
> 

Why did you have to sell it ? Was the battery drain that much worse ? Did you notice heat to much ? Did it break the OS ? 

Why would you recommend a Sabrent without Q?

I was looking at this one: https://www.amazon.de/Sabrent-Rocket-NVMe-Hochleistungs-Solid-State-Laufwerk-SB-ROCKET-512/dp/B07KGMBCKD/ref=sr_1_5?__mk_de_DE=%C3%85M%C3%85%C5%BD%C3%95%C3%91&dchild=1&keywords=Sabrent+Rocket&qid=1602244095&sr=8-5

Would that work ?

> > Ok is it hard to set up ? Its seems worth the effort but the guides are all skeleton like a bit more info what wich part does what and how it fits all together would be nice...
> 
> Only follow "How to install" on stevezhengshiqi/one-key-cpufriend.
> Huge thanks to @stevezhengshiqi for all the work!
> @ManuGithubSteam, you can undervolt you beast Mi Hackintosh Pro, it gives you more battery life, more power and less heat. Check that [here](https://github.com/daliansky/XiaoMi-Pro-Hackintosh/issues/389). I use these values:
> CPU voltage offset: -107mv
> GPU voltage offset: -73mv
> CPU Cache voltage offset: -107mv
OK this is cool stuff.
What Bios Version are you running ?

**Comment by ManuGithubSteam** [comment]
@MarFre22 

Ok i could successfully apply the Bios patched for undervolting, sst and the unlock of the cfg and dvm. I used the 906er bios The Fan Patch from https://github.com/Cr0wTom/Mi-Notebook-Pro-Mods i did not apply because i think it is somewhat included in the 906 Bios and they did not have that EC bin file for the 906 Bios. Do you have experience with the Fan Curve patch ?

I will play with undervolting later. Now i want to get the other improvements going namely NVME fix and the HiDPI

However when i tried to do the NVMEFix Mac OS does not boot anymore (no kernel crash just the apple logo). I think i botched my config.plist.

This is what a wrote into it to enable the NVME fix patch:

in the kernel section on after the bluetooth injection thing.

			<dict>
				<key>Arch</key>
				<string>x86_64</string>
				<key>BundlePath</key>
				<string>NVMeFix.kext</string>
				<key>Comment</key>
				<string>NVME_Fix</string>
				<key>Enabled</key>
				<true/>
				<key>ExecutablePath</key>
				<string></string>
				<key>MaxKernel</key>
				<string></string>
				<key>MinKernel</key>
				<string>16.0.0</string>
				<key>PlistPath</key>
				<string>Contents/Info.plist</string>
			</dict>

**Comment by ManuGithubSteam** [comment]
Hi Steve
> > Does igfxrpsc work on 10.15.7. ?
> 
> No idea since I am not a developer of [WhateverGreen](https://github.com/acidanthera/WhateverGreen). They just mentioned that `Disabled RPS control patch by default due to a bug in 10.15.6 IGPU drivers` in [Changelog.md](https://github.com/acidanthera/WhateverGreen/blob/f31896efc9eb28e0eaad694e9268c7ae5e6f9134/Changelog.md)
> 
I have 10.15.7 with igfxrpsc=1 so far with no problems.
Also thank you for the usb-driver aml it works. I will play around with it.

**Comment by MarFre22** [comment]
> > Assuming your battery it's rock solid yet, I think, battery draining it's more visible without cpufriend.
> ok thanks will try the cpu friend then.
> > I had a 970 evo some time ago, due to using more MacOS than Windows (dualboot) I needed to sell my 970 to buy an Rocket Q - I know 970 evo is a beast (MLC compared QLC, TBW, general speeds) but on Mac Sabrent Rocket Q has better performance due to the Phison controller (Samsung proprietary controllers have constant TRIM crash on Mac -> more heat). After I bought Sabrent Rocket Q, I have much better performance, huge improvement. If you can buy Sabrent Rocket (not Q), it's a most safest choice due to TBW and speeds compared with 970 evo.
> > 
> 
> Why did you have to sell it ? Was the battery drain that much worse ? Did you notice heat to much ? Did it break the OS ? 

The 970 evo is very usable, only it's a bit slow compared with Sabrent on Mac due to controller. Of course, with sabrent you will get a bit less battery drain. The NVMe Samsung's drives are hotter because it gives more performance, but MacOS doesn't extract all performance from it.
With my previous search, I think 960 evo is more MacOS compatible because of an different controller (sorry if I'm wrong, it's the same controller of Apple's proprietary drives).

> Why would you recommend a Sabrent without Q?

Because it uses Phison controllers (MacOS likes that), and they are "4K native" (they have 4Kb sectors) with that MacOS plays very nice (like on Apple's proprietary drives). 
MacOS happy = more performance and more optimization XD


> I was looking at this one: https://www.amazon.de/Sabrent-Rocket-NVMe-Hochleistungs-Solid-State-Laufwerk-SB-ROCKET-512/dp/B07KGMBCKD/ref=sr_1_5?__mk_de_DE=%C3%85M%C3%85%C5%BD%C3%95%C3%91&dchild=1&keywords=Sabrent+Rocket&qid=1602244095&sr=8-5
> 
> Would that work ?

Yes, it will work very well ;)

> OK this is cool stuff.
> What Bios Version are you running ?

906 version. I think there is a newer version, but I'm very good with 906.

**Comment by MarFre22** [comment]
> @MarFre22 
> 
> Ok i could successfully apply the Bios patched for undervolting, sst and the unlock of the cfg and dvm. I used the 906er bios The Fan Patch from https://github.com/Cr0wTom/Mi-Notebook-Pro-Mods i did not apply because i think it is somewhat included in the 906 Bios and they did not have that EC bin file for the 906 Bios. Do you have experience with the Fan Curve patch ?

Fan Patch it's not compatible with 906, because it already has that kind of improvements.

> I will play with undervolting later. Now i want to get the other improvements going namely NVME fix and the HiDPI
>
> However when i tried to do the NVMEFix Mac OS does not boot anymore (no kernel crash just the apple logo). I think i botched my config.plist.
> 
> This is what a wrote into it to enable the NVME fix patch:
> 
> in the kernel section on after the bluetooth injection thing.
> 
> 			<dict>
> 				<key>Arch</key>
> 				<string>x86_64</string>
> 				<key>BundlePath</key>
> 				<string>NVMeFix.kext</string>
> 				<key>Comment</key>
> 				<string>NVME_Fix</string>
> 				<key>Enabled</key>
> 				<true/>
> 				<key>ExecutablePath</key>
> 				<string></string>
> 				<key>MaxKernel</key>
> 				<string></string>
> 				<key>MinKernel</key>
> 				<string>16.0.0</string>
> 				<key>PlistPath</key>
> 				<string>Contents/Info.plist</string>
> 			</dict>
> 
> My whole plist is here: https://pastebin.com/7jg624tu (working, without the change from above)
> 
> **Update**: OK i learnd that MacOS is not so chatty like linux...
> Here is the kernel Panic: https://ibb.co/NFMvQ54

I don't know much about it, I just copy the kext into the clover folder. For real, I don't know if it's active XD

@stevezhengshiqi, sorry for bothering you, it's possible you can give us some clarification about NVMEFIX installation?

**Comment by stevezhengshiqi** [comment]
@MarFre22 
- Download the latest version from https://github.com/acidanthera/NVMeFix/releases
- Put NVMeFix.kext to /EFI/OC/Kexts
- Change https://github.com/daliansky/XiaoMi-Pro-Hackintosh/blob/master/OC/config.plist#L708 to true
- Restart

**Comment by MarFre22** [comment]
> @MarFre22 
> - Download the latest version from https://github.com/acidanthera/NVMeFix/releases
> - Put NVMeFix.kext to /EFI/OC/Kexts
> - Change https://github.com/daliansky/XiaoMi-Pro-Hackintosh/blob/master/OC/config.plist#L708 to true
> - Restart

Thank you very much for the quick response.
Is it any problem in applying it in clover? or do you advisesme to install OC? I'm currently in clover because I'm afraid to lost my windows programs activations.

**Comment by stevezhengshiqi** [comment]
No worry. Clover is good with NVMeFix.

**Comment by ManuGithubSteam** [comment]
> > @MarFre22
> > 
> > * Download the latest version from https://github.com/acidanthera/NVMeFix/releases
> > * Put NVMeFix.kext to /EFI/OC/Kexts
> > * Change https://github.com/daliansky/XiaoMi-Pro-Hackintosh/blob/master/OC/config.plist#L708 to true
> > * Restart
> 
> Thank you very much for the quick response.
> Is it any problem in applying it in clover? or do you advisesme to install OC? I'm currently in clover because I'm afraid to lost my windows programs activations.

Hi MarFe, My activation of Windows stayed the same (active) after installing OC with a generated SMBIOS. This is just my experiance but i think OC will not change the Hardware UUID of another M2 SSD. Its contained in the one SSD or should be even partiton. So it should work....

**Comment by ManuGithubSteam** [comment]
> 
> I don't know much about it, I just copy the kext into the clover folder. For real, I don't know if it's active XD
> 
There is a way to check its in the NVME Repo you can querry some system info and it will show you if it is active. I did not have the time to look into it (prepping for an exam in lie 3 hours....)
But i will latest on monday :-)

**Comment by MarFre22** [comment]
> > > @MarFre22
> > > 
> > > * Download the latest version from https://github.com/acidanthera/NVMeFix/releases
> > > * Put NVMeFix.kext to /EFI/OC/Kexts
> > > * Change https://github.com/daliansky/XiaoMi-Pro-Hackintosh/blob/master/OC/config.plist#L708 to true
> > > * Restart
> > 
> > Thank you very much for the quick response.
> > Is it any problem in applying it in clover? or do you advisesme to install OC? I'm currently in clover because I'm afraid to lost my windows programs activations.
> 
> Hi MarFe, My activation of Windows stayed the same (active) after installing OC with a generated SMBIOS. This is just my experiance but i think OC will not change the Hardware UUID of another M2 SSD. Its contained in the one SSD or should be even partiton. So it should work....

I'm using one 1TB Sabrent Rocket Q ssd with windows and Mac (it's stable, but I always make backups of EFI folder) - again huge thanks to @stevezhengshiqi for the installation wiki! (btw, I found some errors on it about using the program, can you give me the permission to fix it? also I've been thinking for a long time about to give some help in order to improve the wiki section with SSDs topic, undervolt, etc (mainly summarize some importante issues). - it will work? because windows activation it's easy to fix, but other programs like antivirus and some games with have the license bounded the motherboard.

**Comment by MarFre22** [comment]
> 
> > 
> > I don't know much about it, I just copy the kext into the clover folder. For real, I don't know if it's active XD
> > 
> There is a way to check its in the NVME Repo you can querry some system info and it will show you if it is active. I did not have the time to look into it (prepping for an exam in lie 3 hours....)
> But i will latest on monday :-)
> 

Yeah I think so, I will try it ;)
Have a good exam! 💪

**Comment by stevezhengshiqi** [comment]
@MarFre22 That's very kind of you. I will ask @daliansky about adding you as a collaborator.

**Comment by MarFre22** [comment]
> @MarFre22 That's very kind of you. I will ask @daliansky about adding you as a collaborator.

Thank you very much ;)

**Comment by ManuGithubSteam** [comment]
> > Hi MarFe, My activation of Windows stayed the same (active) after installing OC with a generated SMBIOS. This is just my experiance but i think OC will not change the Hardware UUID of another M2 SSD. Its contained in the one SSD or should be even partiton. So it should work....
> 
> I'm using one 1TB Sabrent Rocket Q ssd with windows and Mac (it's stable, but I always make backups of EFI folder) - again huge thanks to @stevezhengshiqi for the installation wiki! (btw, I found some errors on it about using the program, can you give me the permission to fix it? also I've been thinking for a long time about to give some help in order to improve the wiki section with SSDs topic, undervolt, etc (mainly summarize some importante issues). - it will work? because windows activation it's easy to fix, but other programs like antivirus and some games with have the license bounded the motherboard.
>
Hmm i think it might work because Windows has its own bootloader and it gets loaded from OC i belive. So it should work.
Also you could read the UUID from the motherboard and use this one in OC to be sure. The UUID can be read rom windows with this: wmic csproduct get "UUID" (in Powershell)

**Comment by MarFre22** [comment]
> Hmm i think it might work because Windows has its own bootloader and it gets loaded from OC i belive. So it should work.
> Also you could read the UUID from the motherboard and use this one in OC to be sure. The UUID can be read rom windows with this: wmic csproduct get "UUID" (in Powershell)

Oh, thanks! I think you solved my problem!
I'll try it, not now, but when I get some free time

**Comment by ManuGithubSteam** [comment]
Very glad i could help :-) Looking forward to the undervolting article :-)

**Comment by ManuGithubSteam** [comment]
@MarFre22 

What settings did you use in cpufriend ?

Im using 800 mhz now and balanced power. Im saving around 200 Mhz of cpu - so this should save some battery... Just wondering what you are useing as you have more exeriance with the laptop. What are good values ?

Thanks a lot!

**Comment by MarFre22** [comment]
> @MarFre22
> 
>     * Download the latest version from https://github.com/acidanthera/NVMeFix/releases
> 
>     * Put NVMeFix.kext to /EFI/OC/Kexts
> 
>     * Change https://github.com/daliansky/XiaoMi-Pro-Hackintosh/blob/master/OC/config.plist#L708 to true
> 
>     * Restart

I'm trying to find any entries of NVMeFix in Clover config.plist, but I can not find any? What can I do?

**Comment by ManuGithubSteam** [comment]
> > @MarFre22
> > ```
> > * Download the latest version from https://github.com/acidanthera/NVMeFix/releases
> > 
> > * Put NVMeFix.kext to /EFI/OC/Kexts
> > 
> > * Change https://github.com/daliansky/XiaoMi-Pro-Hackintosh/blob/master/OC/config.plist#L708 to true
> > 
> > * Restart
> > ```
> 
> I'm trying to find any entries of NVMeFix in Clover config.plist, but I can not find any? What can I do?
I found them in the OC config. Can they ported back to Clover  ?

**Comment by MarFre22** [comment]
> > I'm trying to find any entries of NVMeFix in Clover config.plist, but I can not find any? What can I do?
> >
> I found them in the OC config. Can they ported back to Clover  ?
> 

It's a bit strange on version 1.4.3 (which has NVMEfix kext) doesn't have any entry related to it. I only found one entry related with TRIM.

**Comment by ManuGithubSteam** [comment]
> > > I'm trying to find any entries of NVMeFix in Clover config.plist, but I can not find any? What can I do?
> > 
> > 
> > I found them in the OC config. Can they ported back to Clover  ?
> 
> @ManuGithubSteam
> It's a bit strange on version 1.4.3 (which has NVMEfix kext) doesn't have any entry related to it. I only found one entry related with TRIM.

hmm i dont use clover so im sadly not the best person to ask for that.
In my wiki i wrote how you can check if it is working, if you have clover loading it.

Also i made a cpufreidn profile. see comment above. Do you know of any way to measure time till battery is empty ? So we could get a better idea how much battery is saved ....

**Comment by MarFre22** [comment]
> @MarFre22 
> 
> What settings did you use in cpufriend ?
> 
> Im using 800 mhz now and balanced power. Im saving around 200 Mhz of cpu - so this should save some battery... Just wondering what you are useing as you have more exeriance with the laptop. What are good values ?
> 
> Thanks a lot!

@ManuGithubSteam 
I'm using 800MHz and balanced performance. With balanced power the CPU never get the Intel turbo boost (limited to some seconds), but if you only need battery life -balanced power is safe choice, but if like to use your CPU for some heavy task not forgetting battery life - balanced performance is the choice.

**Comment by MarFre22** [comment]
> hmm i dont use clover so im sadly not the best person to ask for that.
> In my wiki i wrote how you can check if it is working, if you have clover loading it.
> 
> Also i made a cpufreidn profile. see comment above. Do you know of any way to measure time till battery is empty ? So we could get a better idea how much battery is saved .... 

@ManuGithubSteam 
I know, it's better to change the home to OpenCore XD

The app FruitJuice (see on "very legal" websites), it's a great app for what you need.

**Comment by MarFre22** [comment]
> > hmm i dont use clover so im sadly not the best person to ask for that.
> > In my wiki i wrote how you can check if it is working, if you have clover loading it.
> > 
> > Also i made a cpufreidn profile. see comment above. Do you know of any way to measure time till battery is empty ? So we could get a better idea how much battery is saved .... 
> 
> @ManuGithubSteam 
> I know, it's better to clean the home to OpenCore XD
> 
> The app FruitJuice (see on "very legal" websites), it's a great app for what you need.

**Comment by ManuGithubSteam** [comment]
@MarFre22 
Thank you that is really good advice!
Will rebuild hen. Im thinking of going to 750 mhz just to see if it makes a bit of a difference.

Also im looking forward to you undervoltage article. I read a bit around in the git threads here and it seems to have a lot of moving parts who need to work together. So a good layout would be benificial.

I will work in the meantime on the sd-card-reader. Good news. Its working in a linux vm. It just needs to be made more user friendly. So im working in that direction.

For the SD-Card reader thing i will open a new issue so its cleaner tho :100:

**Comment by MarFre22** [comment]
> @MarFre22 
> Thank you that is really good advice!
> Will rebuild hen. Im thinking of going to 750 mhz just to see if it makes a bit of a difference.
> 
> Also im looking forward to you undervoltage article. I read a bit around in the git threads here and it seems to have a lot of moving parts who need to work together. So a good layout would be benificial.
> 
> I will work in the meantime on the sd-card-reader. Good news. Its working in a linux vm. It just needs to be made more user friendly. So im working in that direction.
> 
> For the SD-Card reader thing i will open a new issue so its cleaner tho :100: 
> 

@ManuGithubSteam 
Yeah, if you have time to improve cpufriend, go for it, you can do it 💪  all the improvements are welcome!

Give me some time please :), you can check the links I sent, I described the solution in these.

SD card friendly compatibility would great ;)

**Comment by stevezhengshiqi** [comment]
> > > I'm trying to find any entries of NVMeFix in Clover config.plist, but I can not find any? What can I do?
> > 
> > 
> > I found them in the OC config. Can they ported back to Clover  ?
> 
> @ManuGithubSteam
> It's a bit strange on version 1.4.3 (which has NVMEfix kext) doesn't have any entry related to it. I only found one entry related with TRIM.

Clover doesn't need any kext entries, you can just put a kext into /EFI/CLOVER/kexts/Other and it will work.

**Comment by MarFre22** [comment]
> > @ManuGithubSteam
> > It's a bit strange on version 1.4.3 (which has NVMEfix kext) doesn't have any entry related to it. I only found one entry related with TRIM.
> 
> Clover doesn't need any kext entries, you can just put a kext into /EFI/CLOVER/kexts/Other and it will work.

@stevezhengshiqi 
Oh well, it's solved, thank you very much for the constant help and support!

**Comment by ManuGithubSteam** [comment]
Hi @MarFre22 

I tried my luck my luck with the undervolting.

As far as i understand it:

Sudo ./voltageshift buildlaunchd -125 0 -125 0 0 0 25.5 999999 25.5 999999 60

Will undervolt but also deactivate the underclocking. As the base clock gows up again (im runnning successfully with 700Mhz btw).

So we can not use the TDP paramters unless we are willing to loose the underclocking we did with cpufriend.

correct ?

This means we can oly use the first three parameters to undervolt like you wrote before.

Also how do i use speedshift ? The script is not on github anymore ...

But it seems the speedshift value have something to do with the TDP Values.

Could you shed some light to this and what parameters do you use in speedshift. How does it fit in the cpufriend side of things and in combination with voltshift ?

Thanks! :-)

**Comment by MarFre22** [comment]
Hi @ManuGithubSteam 
Finally I made the Undervolt Wiki page, hope the community like it. See if it helps you ;)

**Comment by ManuGithubSteam** [comment]
> Hi @ManuGithubSteam
> Finally I made the [Undervolt Wiki page,](https://github.com/daliansky/XiaoMi-Pro-Hackintosh/wiki/Undervolt) hope the community like it. See if it helps you ;)

Hi MarFre22

Thank you for the guide.

I have a couple of questions:

What is the difference between:./voltageshift buildlaunchd -125 0 -125

See here : https://github.com/daliansky/XiaoMi-Pro-Hackintosh/issues/180

And your command.
Can the TDP command be used ?
As it would save more power correct ?

Im using at the moment:
sudo ./voltageshift offset -107 -73 -107 0 0 0 25.5 999999 25.5 999999 60

How do i use the intel speed step technology on the mac? The Pasi script and repo seems to be offline? How did you do that?

Also i belive the ./voltageshift buildlaunchd does the same as your graphical application. Is that so ?

Thanks and thank you for the guide :-)

Did you have a chance to test my sd-card solution ?

**Comment by stevezhengshiqi** [comment]
> Also how do i use speedshift ? The script is not on github anymore ...

https://github.com/daliansky/XiaoMi-Pro-Hackintosh/blob/master/BIOS/DVMT_and_0xE2_fix/speedshift_unlock.cmd

**Comment by ManuGithubSteam** [comment]
> > Also how do i use speedshift ? The script is not on github anymore ...
> 
> https://github.com/daliansky/XiaoMi-Pro-Hackintosh/blob/master/BIOS/DVMT_and_0xE2_fix/speedshift_unlock.cmd

Hi Steve,

I dont the biso patch already. Do i need to do anyting else in Macos to control it ?

Thanks :-)

**Comment by MarFre22** [comment]
> > > Also how do i use speedshift ? The script is not on github anymore ...
> > 
> > https://github.com/daliansky/XiaoMi-Pro-Hackintosh/blob/master/BIOS/DVMT_and_0xE2_fix/speedshift_unlock.cmd
> 
> Hi Steve,
> 
> I dont the biso patch already. Do i need to do anyting else in Macos to control it ?
> 
> Thanks :-)

You must to use the speedshift_unlock.cmd on your windows partition.
Before that take a look [here](https://github.com/Cr0wTom/Mi-Notebook-Pro-Mods).

**Comment by ManuGithubSteam** [comment]
> > > > Also how do i use speedshift ? The script is not on github anymore ...
> > > 
> > > 
> > > https://github.com/daliansky/XiaoMi-Pro-Hackintosh/blob/master/BIOS/DVMT_and_0xE2_fix/speedshift_unlock.cmd
> > 
> > 
> > Hi Steve,
> > I dont the biso patch already. Do i need to do anyting else in Macos to control it ?
> > Thanks :-)
> 
> You must to use the speedshift_unlock.cmd on your windows partition.
> Before that take a look [here](https://github.com/Cr0wTom/Mi-Notebook-Pro-Mods).

lesson learnt! Dont type to fast in a hurry! The bios patch is already applied (did that on windows). As is the undervolting one and undervolting works.
Do i need to do anything else in mac to control the bios patch for the speed step or is the patch itself enough in activating the functionality and it works out of the box ?
Thanks :-)

**Comment by MarFre22** [comment]
> > You must to use the speedshift_unlock.cmd on your windows partition.
> > Before that take a look [here](https://github.com/Cr0wTom/Mi-Notebook-Pro-Mods).
> 
> lesson learnt! Dont type to fast in a hurry! The bios patch is already applied (did that on windows). As is the undervolting one and undervolting works.
> Do i need to do anything else in mac to control the bios patch for the speed step or is the patch itself enough in activating the functionality and it works out of the box ?
> Thanks :-)

I think you are good to go :)
But as I said before, give a try to my guide with TDP (to be safe, only execute the set.command file without the power manager app).
